### PR TITLE
mount /sbin/apparmor_parser if PodSecurityPolicy is enabled

### LINKED ISSUE #2319: mount apparmor_parser as volume for kubelet
when using pod security policy, a volume mount :  `/sbin/apparmor_parser` is missing for kubelet.

This is a show stopper for psp

@jackfrancis @pidah

### PR DESCRIPTION
**Pull request by jackfrancis**
**What this PR does / why we need it**: Fixes #2319

**Release note**:

```
mount /sbin/apparmor_parser if PodSecurityPolicy is enabled
```

### CODE REVIEW

**Comment by jackfrancis** [comment]
@yastij FYI. Could you verify this change and, if possible, test a cluster buildout w/ PSP enabled against this branch? Better yet, could you share your test steps and we'll bake this into standard E2E tests.

**Comment by pidah** [comment]
@jackfrancis /sbin/apparmor_parser is a file

**Comment by jackfrancis** [comment]
@pidah are you saying that we shouldn't be creating a volume mount like #2319 suggests?

**Comment by pidah** [comment]
@jackfrancis sorry I wasn't clear, `ExecStartPre=/bin/mkdir -p /sbin/apparmor_parser` will fail because a file of the same name exists.

**Comment by jackfrancis** [comment]
@pidah thanks

@yastij could you comment on the expected docker mount statements we should apply if PSP is enabled?

File: parts/k8s/artifacts/kuberneteskubelet.service [hunk 1]
```diff
@@ -10,6 +10,7 @@ SuccessExitStatus=143
 ExecStartPre=/bin/bash /opt/azure/containers/kubelet.sh
 ExecStartPre=/bin/mkdir -p /var/lib/kubelet
 ExecStartPre=/bin/mkdir -p /var/lib/cni
+ExecStartPre=/bin/mkdir -p /sbin/apparmor_parser
```
Comment (yastij) [inline parts/k8s/artifacts/kuberneteskubelet.service]: no need add this, as it is already present

**Comment by jackfrancis** [comment]
@yastij @pidah E2E looks green for linux tests, could you both do a quick sanity code review? Thanks!

**Comment by pidah** [comment]
LGTM @jackfrancis

**Comment by arnaudmz** [comment]
What is the point of mounting the file in **rw**?
Our tests showed that **ro** seems enough.

**Comment by yastij** [comment]
@jackfrancis - can you also include this in the next patch release please, as it makes psp unusable with acs-engine ?

**Pull request merged**